flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -960,12 +965,26 @@ public void handleError(final Exception exception) {
 
 		@Override
 		public void releaseResource(InstanceID instanceId) {
-			stopWorker(instanceId);
+			runAsync(new Runnable() {
+				@Override
+				public void run() {
+					for (Map.Entry<ResourceID, WorkerRegistration<WorkerType>> entry : taskExecutors.entrySet()) {
+						if (entry.getValue().getInstanceID().equals(instanceId)) {
+							stopWorker(entry.getKey());
```

Comment:
In the future we should make these ids being composed of each other. Then we should easily obtain the `ResourceID` from the `InstanceID`